### PR TITLE
fix: rm usage of repr for ARC compatibility

### LINKED ISSUE #6: ARC compatibility
Hello! I was testing some Nimble packages with ARC, and found out that your package fails tests with ARC. It turned out that the cause is simple - you're relying in `repr` which behaves differently for ARC.

I suggest you to consider not using repr at all (since it's really only for humans and isn't supposed to be parsed programmatically), or at least fix it for ARC (and ORC which will probably become the default in 1.4), the simplest possible fix being:
```nim
      runedAlphabet.repr[0 .. ^(when defined(gcArc): 1 else: 2)].split("@".runeAtPos(0), 1)[1] &
```
on line 314 in `nbaser.nim`

### PR DESCRIPTION
**Pull request by D-Nice**
Fixes https://github.com/D-Nice/nbaser/issues/6
Also some code cleanup

### CODE REVIEW

**Comment by D-Nice** [comment]
Dude, where's my fuzzer...

**Comment by D-Nice** [comment]
Raison d'être for (most) these force-pushes is in comments of https://github.com/D-Nice/nbaser/commit/2e35169927fb9cdc9e57a1de7d4c52b93e480426
TL;DR a GH actions change (thanks MSFT) broke the previous fuzzer setup, and force-pushed several times to debug it
